tests/generator/test_passthrough.py

```diff
@@ -58,6 +58,83 @@ def test_passthrough_basic(self):
 method=ignore
 '''}, '''[device-netplan.ethernets.NM-87749f1d-334f-40b2-98d4-55db58965f5f]
 match-device=type:ethernet
+managed=1\n\n''')
+
+    def test_passthrough_basic_new_format(self):
+        self.generate('''network:
+  version: 2
+  ethernets:
+    NM-87749f1d-334f-40b2-98d4-55db58965f5f:
+      renderer: NetworkManager
+      match: {}
+      networkmanager:
+        uuid: 87749f1d-334f-40b2-98d4-55db58965f5f
+        name: some NM id
+        passthrough:
+          connection:
+            uuid: 87749f1d-334f-40b2-98d4-55db58965f5f
+            type: ethernet
+            permissions: ""''')
+
+        self.assert_nm({'NM-87749f1d-334f-40b2-98d4-55db58965f5f': '''[connection]
+id=some NM id
+type=ethernet
+uuid=87749f1d-334f-40b2-98d4-55db58965f5f
+#Netplan: passthrough setting
+permissions=
+
+[ethernet]
+wake-on-lan=0
+
+[ipv4]
+method=link-local
+
+[ipv6]
+method=ignore
+'''}, '''[device-netplan.ethernets.NM-87749f1d-334f-40b2-98d4-55db58965f5f]
+match-device=type:ethernet
+managed=1\n\n''')
+
+    def test_passthrough_basic_new_format_with_duplication(self):
+        self.generate('''network:
+  version: 2
+  ethernets:
+    NM-87749f1d-334f-40b2-98d4-55db58965f5f:
+      renderer: NetworkManager
+      match: {}
+      networkmanager:
+        uuid: 87749f1d-334f-40b2-98d4-55db58965f5f
+        name: some NM id
+        passthrough:
+          connection:
+            uuid: 87749f1d-334f-40b2-98d4-55db58965f5f
+            type: ethernet
+            permissions: ""
+          group2:
+            key: a
+            key: b''', skip_generated_yaml_validation=True)
```

Comment:
question: should the result rather be `key=b` here. So that the value defined later wins?

also followup question: what happens if we mix and match the old format?

e.g.
```yaml
passthrough:
  connection:
    name: "another name"
  connection.name: "some name"
 ```

I'd argue the value defined in the new (mapping) format should take precendence. WDYT?

Reply:
hmm yeaaah it probably should be `key=b` in the end. But that's actually a problem in the `handle_generic_map` function. It will error out if it finds the same key with different values (see below). Arguably this logic should be used everywhere else then...

```
$ cat /tmp/fakeroot/etc/netplan/a.yaml
network:
  version: 2
  renderer: NetworkManager
  ethernets:
    renderer: NetworkManager
    enp5s0:
      dhcp4: true
      openvswitch:
        other-config:
          one: 1
          one: 2
$ netplan  get --root-dir /tmp/fakeroot
Command failed: /tmp/fakeroot/etc/netplan/a.yaml:10:11: Error in network definition: duplicate map entry 'one'
          one: 1
          ^
```

For the second question, mixed format is supported. Although, if the key is set using the old style first, the parser will ignore it if it's found again using the new format. The problem is the same described above, but for these cases I'm ignoring the error.

I could create a new `handle_generic_map` only for the passthrough settings... but in this case I suppose the same logic should be used across all the parser and not only here...